tests/components/lamarzocco/test_config_flow.py

```diff
@@ -272,6 +273,69 @@ async def test_reauth_flow(
     assert mock_config_entry.data[CONF_PASSWORD] == "new_password"
 
 
+async def test_reconfigure_flow(
+    hass: HomeAssistant,
+    mock_cloud_client: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    mock_device_info: LaMarzoccoDeviceInfo,
+) -> None:
+    """Testing reconfgure flow."""
+    mock_config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": SOURCE_RECONFIGURE,
+            "unique_id": mock_config_entry.unique_id,
+            "entry_id": mock_config_entry.entry_id,
+        },
+        data=mock_config_entry.data,
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure_confirm"
+
+    result2 = await __do_successful_user_step(hass, result, mock_cloud_client)
+    service_info = get_bluetooth_service_info(
+        mock_device_info.model, mock_device_info.serial_number
+    )
+
+    with (
+        patch(
+            "homeassistant.components.lamarzocco.config_flow.LaMarzoccoLocalClient.validate_connection",
+            return_value=True,
+        ),
+        patch(
+            "homeassistant.components.lamarzocco.config_flow.async_discovered_service_info",
+            return_value=[service_info],
+        ),
+    ):
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {
+                CONF_HOST: "192.168.1.1",
+                CONF_MACHINE: mock_device_info.serial_number,
+            },
+        )
+    await hass.async_block_till_done()
+
+    assert result3["type"] is FlowResultType.FORM
+    assert result3["step_id"] == "bluetooth_selection"
+
+    result4 = await hass.config_entries.flow.async_configure(
+        result3["flow_id"],
+        {CONF_MAC: service_info.address},
+    )
+
+    assert result4["type"] is FlowResultType.ABORT
+    assert result4["reason"] == "reconfigure_successful"
+
+    entry = hass.config_entries.async_get_entry(mock_config_entry.entry_id)
+    assert entry
+    assert entry.title == "My LaMarzocco"
+    assert entry.data == {**mock_config_entry.data, CONF_MAC: service_info.address}
```

Comment:
```suggestion
    assert mock_config_entry.title == "My LaMarzocco"
    assert mock_config_entry.data == {**mock_config_entry.data, CONF_MAC: service_info.address}
```
Rather write out the expected data I think, because currently you're overwriting nothing as the config entries are the same object

Reply:
That's what the current test is doing yes, you're comparing it against itself